### ISSUE TITLE
Use case analysis:  ZIM Wikipedia snapshots from Kiwix

### ISSUE THREAD

**Issue by lidel** [lidel/zim2ipfs#1]
### About Kiwix 

> Wherever you go, you can browse Wikipedia, read books from the Gutenberg Library, or watch TED talks and much more – even if you don’t have an Internet connection.
> Kiwix is a non-profit. All code is free and open-source.
> – https://www.kiwix.org/en/

Kiwix builds ZIM archives with Wikipedia snapshots (https://farm.openzim.org/pipeline/filter-doing)  and provides them at  http://download.kiwix.org/zim/wikipedia/

### Value added by putting ZIM on IPFS

Right now, Kiwix is able to store only two last months of each language – putting them on IPFS allows us to keep them around for longer

###  How to do it?

I've identified two high level options for putting ZIM on IPFS:

- (A) Do IPFS import (chunking) and initial "seeding" yourselves, use pinning services for seeding
- (B) HTTP POST data to pinning service, and let it do both IPFS import, chunking and seeding.

The key differences are:

**(A)**  For the duration of initial IPFS "seeding", Kiwix runs their own, temporary IPFS node (https://hub.docker.com/r/ipfs/go-ipfs / https://dist.ipfs.io/#go-ipfs). This node does the IPFS import: chunking and hashing of data locally. It can use [Filestore](https://github.com/ipfs/go-ipfs/blob/master/docs/experimental-features.md#ipfs-filestore) so .zim file from regular filesystem is used directly (no data duplication) or [Urlstore](https://github.com/ipfs/go-ipfs/blob/master/docs/experimental-features.md#ipfs-urlstore) if it is exposed over HTTP on other box – this way existing storage is re-used.

When import is finished, a CID (content identifier) is returned. Then we send API call to a remote pinning service to pin this specific CID – they will fetch data from Kiwix node, and after they confirm to have a full copy, Kiwix can shut down IPFS node and stop the initial seeding – from now on, remote services will take care of providing mentioned CID to the IPFS swarm.

- Pros:
  - we control IPFS import end-to-end, and get CID we can trust was not modified in any way.
  - can pin at multiple places at the same time, services fetch data from our IPFS node and each other, resume is easy (only missing blocks are requested)
- Cons:
  - bit more setup upfront (but I am determined to help with this, hands-on)
  - hashing and chunking is CPU-intensive (could matter if Kiwix is billed for CPU time – but we could figure this out, PL will be happy to cover the cost)

**(B)** Kiwix does not run IPFS node, when .zim is ready we send it over HTTP to https://estuary.tech/ and get CID in return. Then we can use this CID to pin data to https://web3.storage/ as well.
- Pros:
  - easy setup upfront, low CPU cost (HTTP client uploads data, chunking happens on external service)
- Cons:
  - you need to trust the first remote service that will produce the CID (ok if we use https://estuary.tech/ – it is run by PL folks, but this may change in the future)
  - afaik there is no smart resume – 80GB .ZIM would have to be sent from the beginning on retry
    - (this is something to confirm)

### tl;dr

Personally, I think (A) is more future-proof, as the service used in (B) could (in theory) go away in a few years, and (A) is self-hosted, uses open protocols end-to-end, and multiple pinning services can be used in parallel. 

I'd like to  spend some time during ongoing hack week to figure out the details for (A) – see https://github.com/lidel/zim2ipfs/pull/2
